src/neuroconv/basedatainterface.py

```diff
@@ -116,7 +115,7 @@ def add_to_nwbfile(self, nwbfile: NWBFile, **conversion_options) -> None:
 
     def run_conversion(
         self,
-        nwbfile_path: Optional[str] = None,
+        nwbfile_path: Union[str, Path],
```

Comment:
Do we actually support `Path` here generally yet? IDK if PyNWB/HDMF/HDMF-Zarr want `str` only at IO time

Reply:
I have always assumed we do. PYNWB does not but I am less sure about zarr. Let's revert to restrictive then just out of caution.

Reply:
Done.